Rework the tutorial level and various small fixes

Dennisbonke opened the pull request:
This PR adds the correct textures to the level, introduces basic obstacles, and does some finalizing on it. Also includes some rework in the editor, a complete rework of the level layout and a change of the background color to pitch black.

OPA-41, still misses vent tiles but those are blocked on the implementation of OPA-34 and OPA-102.
Also misses player textures, blocked on OPA-92 for enemies and OPA-116, OPA-118, OPA-119 and OPA-142 for Adam.
Does not include UI textures, partially blocked on OPA-42 and OPA-43, fully blocked on OPA-129, OPA-130 and OPA-131.

lokiy999 reviewed (approved):
LGTM

Westlanderz commented on Assets/Scripts/Core/Tilemap.cs:
```diff
@@ -66,20 +66,29 @@ public class SaveObject {
+			public bool cover;
 			// Holds if the tile can be walked over.
-			public bool walkable { get; protected set; }
+			public bool walkable;
```
See previous comment.

Dennisbonke replied:
Prefabs need to be able to set this.

Westlanderz commented on Assets/Scripts/Core/Tilemap.cs:
```diff
@@ -142,21 +151,15 @@ public class SaveObject {
 					this.hitChanceModifier = floorHitChanceModifier;
 					this.cover = false;
-				} else if(nodeSprite == NodeSprite.WALL) {
+				} else if(nodeSprite == NodeSprite.WALL_FULL_TEXTURE || nodeSprite == NodeSprite.WALL_BACK_TEXTURE || nodeSprite == NodeSprite.WALL_FULL_MIDDLE_TEXTURE || nodeSprite == NodeSprite.NONE || nodeSprite == NodeSprite.DOOR_TOP || nodeSprite == NodeSprite.DOOR_MIDDLE || nodeSprite == NodeSprite.DOOR_BOTTOM) {
```
A bit of a long OR chain, maybe change this to multiple lines for readability.

Dennisbonke replied:
Yep, not wrong, I'll split it up.

Westlanderz reviewed (commented):
See the comments plus why is white_1x1 changing/ moving in the file structure as I added that file already for the health bar system.

Dennisbonke commented on Assets/Scripts/Core/Tilemap.cs:
```diff
@@ -66,20 +66,29 @@ public class SaveObject {
 			}
 			// Holds the amount of cover this tile gives.
-			public bool cover { get; private set; }
+			public bool cover;
```
Prefabs need to be able to set this.

Dennisbonke commented:
> See the comments plus why is white_1x1 changing/ moving in the file structure as I added that file already for the health bar system.

It isn't moving, but your file had the file perms 0755 `(rwxr-xr-x)` and I changed them back to the much more sane 0644 `(rw-r--r--)`.

Westlanderz commented:
> > See the comments plus why is white_1x1 changing/ moving in the file structure as I added that file already for the health bar system.
> 
> It isn't moving, but your file had the file perms 0755 `(rwxr-xr-x)` and I changed them back to the much more sane 0644 `(rw-r--r--)`.

Interesting as either all files by me should have that, or it's because of copying it from code monkey. Either way that's fine then but please check the other files as well, thanks in advance.

Dennisbonke commented:
> > > See the comments plus why is white_1x1 changing/ moving in the file structure as I added that file already for the health bar system.
> > 
> > 
> > It isn't moving, but your file had the file perms 0755 `(rwxr-xr-x)` and I changed them back to the much more sane 0644 `(rw-r--r--)`.
> 
> Interesting as either all files by me should have that, or it's because of copying it from code monkey. Either way that's fine then but please check the other files as well, thanks in advance.

Did a check of all files in `Assets`, changed everything back to 0644.

Westlanderz reviewed (approved):
LGTM now

Westlanderz reviewed (approved):
LGTM now again
